src/jit/codegencommon.cpp

```diff
@@ -3429,6 +3431,19 @@ void CodeGen::genReportEH()
 
         CORINFO_EH_CLAUSE_FLAGS flags = ToCORINFO_EH_CLAUSE_FLAGS(HBtab->ebdHandlerType);
 
+        if (isCoreRTABI && (XTnum > 0))
+        {
+            // For CoreRT, CORINFO_EH_CLAUSE_SAMETRY flag means that the current clause covers same
+            // try block as the previous one. The runtime cannot reliably infer this information from
+            // native code offsets because of different try blocks can have same offsets. Alternative
+            // solution to this problem would be inserting extra nops to ensure that different try
+            // blocks have different offsets.
+            if (EHblkDsc::ebdIsSameTry(HBtab, HBtab - 1))
+            {
+                flags = (CORINFO_EH_CLAUSE_FLAGS)(flags | CORINFO_EH_CLAUSE_SAMETRY);
```

Comment:
Thanks - I have added assert for `HasCatchHandler()`.

I will think about good asserts in the CoreRT side. The runtime won't actually see this bit as is. The whole table is compressed with variable length encoding, and there is no natural place for this bit in the compressed form. [My current implementation](https://github.com/dotnet/corert/pull/2288/files) is inserting empty clauses in the compressed form if it is necessary to explicitly separate runs with the same offset.